### ISSUE TITLE
When Importing Multiple Static Meshes or a Scene, Objects Do Not Keep Their Locations/Scales in UE4?

### ISSUE THREAD

**Issue by KingKong19100** [xavier150/Blender-For-UnrealEngine-Addons#107]
I am not sure if this is a flaw with the importer, but whenever I try to import multiple static meshes (they have precise locations within my scene), they do not import to UE4 correctly. They lose their location from Blender to UE4, so is there a way to counteract this? I can join the objects all as one in Blender and export to UE4, but it still will lose the location and scale when in UE4 and I would have to manually place them, which is quite difficult. 

In summary; When importing multiple static meshes and/or skeletal animations, these meshes all lose their scale and locations. Is there a way to fix this issue? Thank you so much!

**Comment by xavier150** [comment]
Hey @KingKong19100 
That because my addon use the object origin as center. You can uncheck "Move to center" in object advanced property for fix that. Do this on the all objects ^^

**Comment by KingKong19100** [comment]
> Hey @KingKong19100 !
> 
> That because my addon use the object origin as center. That used for modular assets.
> 
> You can uncheck "Move to center" in object advanced property for fix that. ^^
> Do this on the all objects you want export.
> 
> Edit: But object should not lose the scale. If object lose the scale that maybe a bug.
> If you confirm object lose scale you can send me the file?
> 
> (I will able to check when I back to home in 15 days)
Thank you so much! The thing is, it worked for some static meshes, but when it comes to my skeletal meshes, there is an issue with the characters that are animated (Godzilla and Rodan) who are not exporting with the exact scale and location. There was an option that asked to set to UE4 unit scale, and so I clicked on that option, but I did notice that my skeletal meshes are importing incorrectly. So my entire scene in UE4 is very very small.

The static meshes that DO import correctly in the same locations do have an issue, though, and this is because some of my static meshes are animated and keyframes as objects, but when I import them into UE4 they do not have the same animations.

In short; My skeletal mesh characters are importing with wrong scale and location (even when I uncheck "Move to Center"), my entire UE4 scene is extremely small (most likely due to the unit scale), and static meshes that do have animation on them such as the "trees" in my scene are not importing to UE4 with the same animations.  Is there a fix to these issues? Thank you so much!

Here is the Blend File if you need it!

**Comment by xavier150** [comment]
I see, you should speak about the unit scale. By default Blender use a scene Unit scale at 1.0 but Unreal Engine Unit scale at 0.01.

My addon will show a warning about this and proposes to set the scene unit scale at 0.01 like in Unreal

For fix the scale:
- You can keep unit scale at 0.01 and rescale x100 the all assets.
- Or re change the scene unit scale at 1.0.

I recommend rechange unit scale for not have to rescale the animations. 

In blender set Unit scale at 1.0
![tdv97.png](https://user-images.githubusercontent.com/7216958/124350672-3f937c00-dbf6-11eb-8355-00c2a2323b4d.png)

-----

Sadly Static Mesh animation can't be imported. I recommend remove the animations keyframes on Static Mesh in Blender, and animate your Static Mesh in Unreal sequencer. (Don't use the same sequencer than camera for be able to reimport Cameras without lost your Static Mesh animation in sequencer.)

-----
Change the unit scale at 1.0 and tell me if that work better.

Thanks for the file, I will check when I come back ^^

**Comment by KingKong19100** [comment]
> I see, you should speak about the unit scale. By default Blender use a scene Unit scale at 1.0 but Unreal Engine Unit scale at 0.01.
> 
> My addon will show a warning about this and proposes to set the scene unit scale at 0.01 like in Unreal
> 
> For fix the scale:
> 
> * You can keep unit scale at 0.01 and rescale x100 the all assets.
> * Or re change the scene unit scale at 1.0.
> 
> I recommend rechange unit scale for not have to rescale the animations.
> 
> In blender set Unit scale at 1.0
> ![tdv97.png](https://user-images.githubusercontent.com/7216958/124350672-3f937c00-dbf6-11eb-8355-00c2a2323b4d.png)
> 
> Sadly Static Mesh animation can't be imported. I recommend remove the animations keyframes on Static Mesh in Blender, and animate your Static Mesh in Unreal sequencer. (Don't use the same sequencer than camera for be able to reimport Cameras without lost your Static Mesh animation in sequencer.)
> 
> Change the unit scale at 1.0 and tell me if that work better.
> 
> Thanks for the file, I will check when I come back ^^


Hey, thank you so much! It worked for both of my characters and the landscape! But when I keyframe locations of armatures in object mode, this doesn't import to UE4. Is there a way to turn this on? I have characters moving from one end to another and it isn't being imported to UE4.

**Comment by xavier150** [comment]
Hey @KingKong19100 I'm back to work!
Sadly animations in object more, are not supported, you should animate your character with rig bones.

**Comment by KingKong19100** [comment]
> Hey @KingKong19100 I'm back to work!
> Sadly animations in object more, are not supported, you should animate your character with rig bones.

Ah, I understand. In the future will you be able to add this feature?

**Comment by xavier150** [comment]
I don't think unfortunately :c This may be very complex to do unnecessarily
